students/JerryH/Lesson07/html_render.py

```diff
@@ -0,0 +1,120 @@
+class Element:
+
+    # Class Attributes
+    tag = ""
+    indent = "  "
+
+    def __init__(self, content=None, **style):
+        self.content = [content] if content else []
+        self.styles = style
+
+
+    def append(self, words):
+        self.content.append(words)
+
+
+    def render(self, fileout, cur_ind=""):
+        fileout.write(cur_ind + "<{}".format(self.tag))
+
+        for key, value in self.styles.items():
+            fileout.write(' {}="{}"'.format(key,value))
+        fileout.write(">\n")
+
+        for item in self.content:
+            if isinstance(item, Element):
+                item.render(fileout, cur_ind + self.indent)
+            else: # output the content
+                fileout.write(cur_ind + self.indent + item)
+                fileout.write("\n")
+                
+        fileout.write(cur_ind + "</{}>\n".format(self.tag))
+
+
+class Html(Element):
+    tag = 'html'
+
+
+    def render(self, fileout, cur_ind=""):
+        fileout.write("<!DOCTYPE html>\n")
+        Element.render(self, fileout, cur_ind="")
+
+
+class Body(Element):
+    tag = 'body'
+
+
+class P(Element):
+    tag = 'p'
+
+
+class Head(Element):
+    tag = 'head'
+
+
+class SelfClosingTag(Element):
+
+    def __init__(self, **style):
+        ''' SelfClosingTag no content required '''
```

Comment:
styling suggestion: doc strings should use triple double quoted strings `"""docstring"""`